rust/src/wrapper/assigned_numbers/services.rs

```diff
@@ -14,69 +14,298 @@
-        (0x183B, "Binary Sensor"),
-        (0x183C, "Emergency Configuration"),
-        (0x183E, "Physical Activity Monitor"),
-        (0x1843, "Audio Input Control"),
-        (0x1844, "Volume Control"),
-        (0x1845, "Volume Offset Control"),
-        (0x1846, "Coordinated Set Identification Service"),
-        (0x1847, "Device Time"),
-        (0x1848, "Media Control Service"),
-        (0x1849, "Generic Media Control Service"),
-        (0x184A, "Constant Tone Extension"),
-        (0x184B, "Telephone Bearer Service"),
-        (0x184C, "Generic Telephone Bearer Service"),
-        (0x184D, "Microphone Control"),
+    pub static ref SERVICE_IDS: collections::HashMap<Uuid16, NamedUuid> = [
+        (0x1800_u16, GENERIC_ACCESS),
+        (0x1801, GENERIC_ATTRIBUTE),
+        (0x1802, IMMEDIATE_ALERT),
+        (0x1803, LINK_LOSS),
+        (0x1804, TX_POWER),
+        (0x1805, CURRENT_TIME),
+        (0x1806, REFERENCE_TIME_UPDATE),
+        (0x1807, NEXT_DST_CHANGE),
+        (0x1808, GLUCOSE),
+        (0x1809, HEALTH_THERMOMETER),
+        (0x180A, DEVICE_INFORMATION),
+        (0x180D, HEART_RATE),
+        (0x180E, PHONE_ALERT_STATUS),
+        (0x180F, BATTERY),
+        (0x1810, BLOOD_PRESSURE),
+        (0x1811, ALERT_NOTIFICATION),
+        (0x1812, HUMAN_INTERFACE_DEVICE),
+        (0x1813, SCAN_PARAMETERS),
+        (0x1814, RUNNING_SPEED_AND_CADENCE),
+        (0x1815, AUTOMATION_IO),
+        (0x1816, CYCLING_SPEED_AND_CADENCE),
+        (0x1818, CYCLING_POWER),
+        (0x1819, LOCATION_AND_NAVIGATION),
+        (0x181A, ENVIRONMENTAL_SENSING),
+        (0x181B, BODY_COMPOSITION),
+        (0x181C, USER_DATA),
+        (0x181D, WEIGHT_SCALE),
+        (0x181E, BOND_MANAGEMENT),
+        (0x181F, CONTINUOUS_GLUCOSE_MONITORING),
+        (0x1820, INTERNET_PROTOCOL_SUPPORT),
+        (0x1821, INDOOR_POSITIONING),
+        (0x1822, PULSE_OXIMETER),
+        (0x1823, HTTP_PROXY),
+        (0x1824, TRANSPORT_DISCOVERY),
+        (0x1825, OBJECT_TRANSFER),
+        (0x1826, FITNESS_MACHINE),
+        (0x1827, MESH_PROVISIONING),
+        (0x1828, MESH_PROXY),
+        (0x1829, RECONNECTION_CONFIGURATION),
+        (0x183A, INSULIN_DELIVERY),
+        (0x183B, BINARY_SENSOR),
+        (0x183C, EMERGENCY_CONFIGURATION),
+        (0x183E, PHYSICAL_ACTIVITY_MONITOR),
+        (0x1843, AUDIO_INPUT_CONTROL),
+        (0x1844, VOLUME_CONTROL),
+        (0x1845, VOLUME_OFFSET_CONTROL),
+        (0x1846, COORDINATED_SET_IDENTIFICATION),
+        (0x1847, DEVICE_TIME),
+        (0x1848, MEDIA_CONTROL),
+        (0x1849, GENERIC_MEDIA_CONTROL),
+        (0x184A, CONSTANT_TONE_EXTENSION),
+        (0x184B, TELEPHONE_BEARER),
+        (0x184C, GENERIC_TELEPHONE_BEARER),
+        (0x184D, MICROPHONE_CONTROL),
     ]
     .into_iter()
     .map(|(num, name)| (Uuid16::from_le_bytes(num.to_le_bytes()), name))
     .collect();
 }
+
+pub const GENERIC_ACCESS: NamedUuid = NamedUuid::new(
+    Uuid16::from_be_bytes(0x1800_u16.to_be_bytes()),
```

Comment:
nit: would it make sense to pull these values out of the hashmap, so there is one source of truth?

Reply:
Good point; that's an awkward vestige from the former structure. I think it will be cleaner to construct the hashmap from the NamedUuid instances, but either way one source of truth.